python/cudf/cudf/core/frame.py

```diff
@@ -1387,10 +1423,8 @@ def _drop_na_rows(
         diff = set(subset) - set(self._data)
         if len(diff) != 0:
             raise KeyError(f"columns {diff} do not exist")
-        subset_cols = [
-            name for name, col in self._data.items() if name in subset
-        ]
-        if len(subset_cols) == 0:
+
+        if len(subset) == 0:
```

Comment:
Note: at this point `subset` is guaranteed to be a true "subset" columns of the frame, there's no need to construct the intersection of the columns in the frame and the names specified in `subset` to check whether the the "to-drop-list" is null.